### ISSUE TITLE
Multi-tenant Unable to  create/use specifications in newly created tenant so can't create device

### ISSUE THREAD

**Issue by garysmi2** [sitewhere/sitewhere#28]
Hi Derek, myself again.

I'll try and make my bug reports a bit more formal and helpful too yourself.

Using SiteWhere 1.6.0 with the provided sitewhere.war

Pre-Steps: 

- deleted sitewhere and sitewhere tenant databases from mongod
- deleted the contents of the sitewhere folder containing the default and other created tenants

**Issue**

1. Log in as admin and create and activate a new tenant.
2. Logout
3. Log back in again and select the newly created tenant
4. Try and create a device - in the specification tab the specification list is empty.
5. Try and create a new specification using either the default "Asset Provider - Default Device Management" or one created by yourself.
6. Fill in the fields and click the create button.

**Expected:**
A new specification is created and appears in the list.

**Actual:**
List remains empty and there is no visual feedback.

Two browser requests are generated as the result of clicking on create button. The last one fails:

http://localhost:8080/sitewhere/api/specifications?includeAsset=true&tenantAuthToken=nimbustest&take=15&skip=0&page=1&pageSize=15

**Params:**

- includeAsset	true
- page	1
- pageSize	15
- skip	0
- take	15
- tenantAuthToken	nimbustest

**Response:** 

_400 Bad Request  (removed most of the irrelevant html)_

    <u>InvalidAssetReferenceId:Asset reference not found.</u></p><p><b>description</b> <u>The
     request sent by the client was syntactically incorrect.</u>

I've attached partial log too:

[catalina.txt](https://github.com/sitewhere/sitewhere/files/129740/catali

**Comment by derekadams** [comment]
Hi Gary,

Thanks for including the detailed report. We were able to reproduce the problem and have created a Jira issue:

https://sitewhere.atlassian.net/browse/SITEWHERE-277

Thanks,
Derek

**Comment by derekadams** [comment]
This was caused by the last-minute addition of a Node-RED asset for showing off the new integration. We added the asset to the tenants/default/assets/device-assets.xml, but not the tenant-template/assets/device-assets.xml. If you copy the one from the default tenant into the template, delete the tenant databases and tenant subfolders, then restart, the problem should be fixed. I guess we got a little aggressive with last-minute additions. Sorry for the issues.
